metadata-db-app/int-test/cmr/metadata_db/int_test/concepts/subscription_save_test.clj

```diff
@@ -0,0 +1,88 @@
+(ns cmr.metadata-db.int-test.concepts.subscription-save-test
+  "Contains integration tests for saving subscriptions. Tests saves with various configurations including
+  checking for proper error handling."
+  (:require
+   [clojure.test :refer :all]
+   [cmr.common.util :refer (are3)]
+   [cmr.metadata-db.int-test.concepts.concept-delete-spec :as cd-spec]
+   [cmr.metadata-db.int-test.concepts.concept-save-spec :as c-spec]
+   [cmr.metadata-db.int-test.concepts.utils.interface :as concepts]
+   [cmr.metadata-db.int-test.utility :as util]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Fixtures & one-off utility functions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(use-fixtures :each (util/reset-database-fixture {:provider-id "REG_PROV" :small false}))
+
+(defmethod c-spec/gen-concept :subscription
+  [_ _ uniq-num attributes]
+  (concepts/create-concept :subscription "CMR" uniq-num attributes))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Tests
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest save-subscription-test
+  (testing "basic save"
+    (let [concept (c-spec/gen-concept :subscription "CMR" 1 {})]
+      (c-spec/save-concept-test concept 201 1 nil)
+      (testing "save again"
+        (c-spec/save-concept-test concept 201 2 nil))))
+
+  (testing "save after delete"
+    (let [concept (c-spec/gen-concept :subscription "CMR" 1 {})
+          {:keys [email-address concept-id revision-id metadata]} (util/save-concept concept)
+          delete-response (util/delete-concept concept-id)]
+      (is (= 201 (:status delete-response)))
+      (is (= (inc revision-id) (:revision-id delete-response)))
+      (c-spec/save-concept-test concept 201 (+ revision-id 2) nil))))
+
+(deftest save-subscription-failures-test
+  (testing "saving invalid subscription"
+    (are3 [subscription exp-status exp-errors]
+          (let [{:keys [status errors]} (util/save-concept subscription)]
+            (is (= exp-status status))
+            (is (= (set exp-errors) (set errors))))
+          
+          "subscription associated with non system-level provider"
+          (assoc (concepts/create-concept :subscription "CMR" 2) :provider-id "REG_PROV")
+          422
+          ["Subscription could not be associated with provider [REG_PROV]. Subscription is system level entity."])))
```

Comment:
I don't know that expecting the user to give provider-id as a param and writing an error message specifically for that case is better than just having a generic error message for unsupported param name.

Reply:
This is how things are done for variables, services, humanizers, where "CMR" is the provider and I think they do that to fit in the existing mechanism, and I'm doing the same.

Reply:
Ok